src/gui/qgsattributeform.cpp

```diff
@@ -533,6 +539,168 @@ void QgsAttributeForm::onAttributeChanged( const QVariant& value )
       break;
     }
   }
+
+  if ( eww->layer()->editFormConfig()->notNull( eww->fieldIdx() ) )
+  {
+    QLabel* buddy = mBuddyMap.value( eww->widget() );
+
+    if ( buddy )
+    {
+      if ( !buddy->property( "originalText" ).isValid() )
+        buddy->setProperty( "originalText", buddy->text() );
+
+      QString text = buddy->property( "originalText" ).toString();
+
+      if ( value.isNull() )
+      {
+        // not good
+#if QT_VERSION >= 0x050000
+        buddy->setText( QString( "%1<font color=\"red\">❌</font>" ).arg( text ) );
```

Comment:
I think it's more readable this way. And I don't expect that to cause troubles (unless somebody edits the file with a non-utf8 compatible editor)